services/provider/server/consumer.go

```diff
@@ -36,73 +30,35 @@ func newConsumerManager(ctx context.Context, cl client.Client, namespace string)
 		return nil, fmt.Errorf("failed to list storage consumers. %v", err)
 	}
 
-	nameByTicket := map[string]string{}
 	nameByUID := map[types.UID]string{}
 
 	for _, consumer := range consumers.Items {
 		nameByUID[consumer.UID] = consumer.Name
-
-		if ticket, ok := consumer.GetAnnotations()[TicketAnnotation]; ok {
-			nameByTicket[ticket] = consumer.Name
-		}
 	}
 
 	return &ocsConsumerManager{
-		client:       cl,
-		namespace:    namespace,
-		nameByTicket: nameByTicket,
-		nameByUID:    nameByUID,
+		client:    cl,
+		namespace: namespace,
+		nameByUID: nameByUID,
 	}, nil
 }
 
-// Create creates a new storageConsumer resource, updates the consumer cache and returns the storageConsumer UID
-func (c *ocsConsumerManager) Create(ctx context.Context, onboard ifaces.StorageClientOnboarding, storageQuotaInGiB int) (string, error) {
-	ticket := onboard.GetOnboardingTicket()
-	name := onboard.GetConsumerName()
-	c.mutex.RLock()
-	if _, ok := c.nameByTicket[ticket]; ok {
-		c.mutex.RUnlock()
-		klog.Warning("onboarding ticket already in use")
-		return "", errTicketAlreadyExists
-	}
-	c.mutex.RUnlock()
-
-	consumerObj := &ocsv1alpha1.StorageConsumer{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: c.namespace,
-			Annotations: map[string]string{
-				TicketAnnotation: ticket,
-			},
-		},
-		Spec: ocsv1alpha1.StorageConsumerSpec{
-			Enable:            false,
-			StorageQuotaInGiB: storageQuotaInGiB,
-		},
-		Status: ocsv1alpha1.StorageConsumerStatus{
-			Client: ocsv1alpha1.ClientStatus{
-				OperatorVersion: onboard.GetClientOperatorVersion(),
-			},
-		},
-	}
-
-	err := c.client.Create(ctx, consumerObj)
-	if err != nil {
-		if kerrors.IsAlreadyExists(err) {
-			klog.Warningf("storageConsumer %q already exists", name)
-			return "", err
-		}
-		return "", fmt.Errorf("failed to create storageConsumer resource %q. %v", consumerObj.Name, err)
-	}
-
+func (c *ocsConsumerManager) OnboardClient(ctx context.Context, consumer *ocsv1alpha1.StorageConsumer) (string, error) {
```

Comment:
What is the difference between this new one and enable?

Reply:
my mistake I was fixated w/ OnboardClient and total forgot EnableStorageConsumer.

Reply:
fixed.